detection_rules/packaging.py

```diff
@@ -82,52 +83,83 @@ def manage_versions(rules, current_versions=None, exclude_version_update=False,
+
     # update the document with the new rules
-    if new_rules or changed_rules:
+    if new_rules or changed_rules or newly_deprecated:
         if verbose:
             click.echo('Rule hash changes detected!')
 
         if save_changes:
-            current_versions.update(new_rules if add_new else {})
-            current_versions = OrderedDict(sorted(current_versions.items(), key=lambda x: x[1]['rule_name']))
+            if changed_rules or (new_rules and add_new):
+                current_versions.update(new_rules if add_new else {})
+                current_versions = OrderedDict(sorted(current_versions.items(), key=lambda x: x[1]['rule_name']))
+
+                with open(RULE_VERSIONS, 'w') as f:
+                    json.dump(current_versions, f, indent=2, sort_keys=True)
 
-            with open(RULE_VERSIONS, 'w') as f:
-                json.dump(current_versions, f, indent=2, sort_keys=True)
+                if verbose:
+                    click.echo('Updated version.lock.json file')
+
+            if newly_deprecated:
+                with open(RULE_DEPRECATIONS, 'w') as f:
+                    json.dump(sorted(OrderedDict(rule_deprecations)), f, indent=2, sort_keys=True)
 
                 if verbose:
-                    click.echo('Updated version.lock.json file with:')
+                    click.echo('Updated deprecated_rules.json file')
         else:
             if verbose:
-                click.echo('run `build-release --update-version-lock` to update the version.lock.json file')
+                click.echo('run `build-release --update-version-lock` to update the version.lock.json and '
+                           'deprecated_rules.json files')
 
         if verbose:
             if changed_rules:
-                click.echo(' - {} changed rule version(s)'.format(len(changed_rules)))
+                click.echo(f' - {len(changed_rules)} changed rule version(s)')
             if new_rules:
-                click.echo(' - {} new rule version addition(s)'.format(len(new_rules)))
+                click.echo(f' - {len(new_rules)} new rule version addition(s)')
+            if newly_deprecated:
+                click.echo(f' - {len(newly_deprecated)} newly deprecated rule(s)')
 
-    return changed_rules, new_rules.keys()
+    return changed_rules, list(new_rules), newly_deprecated
 
 
 class Package(object):
     """Packaging object for siem rules and releases."""
 
-    def __init__(self, rules, name, release=False, current_versions=None, min_version=None, max_version=None,
-                 update_version_lock=False):
+    def __init__(self, rules, name, deprecated_rules=None, release=False, current_versions=None, min_version=None,
+                 max_version=None, update_version_lock=False):
         """Initialize a package."""
         self.rules = [r.copy() for r in rules]  # type: list[Rule]
         self.name = name
+        self.deprecated_rules = [r.copy() for r in deprecated_rules or []]  # type: list[Rule]
```

Comment:
I think it is better to pass these as a separate list, rather than parse them out of the rules, so that the filter does _not_ apply to these, when building `from_config`, in order to track deprecations more consistently

Reply:
is it necessary to get a full copy of the rule? wondering, because it's an expensive operation

Reply:
They are processed in the same way the prod rules are, so I would imagine whatever the case, applies to both lists